waltz-service/src/main/java/com/khartec/waltz/service/data_flow_decorator/DataFlowDecoratorService.java

```diff
@@ -173,6 +181,14 @@ public int deleteAllDecoratorsForFlowIds(List<Long> flowIds) {
         LogicalFlow flow = dataFlowDao.findByFlowId(flowId);
         dataTypeUsageService.recalculateForApplications(flow.source(), flow.target());
         audit("Added", decoratorReferences, flow, username);
+
+        attestationService.create(ImmutableAttestation.builder()
+                .entityReference(EntityReference.mkRef(LOGICAL_DATA_FLOW, flowId))
+                .attestationType(AttestationType.IMPLICIT)
+                .attestedBy(username)
+                .comments("Implicit due to data type decoration")
+                .build(), username);
+
```

Comment:
Lots of boilerplate repeated in this pull request.

Perhaps a helper method in the service like:
```
attestationService.implicitlyAttest(
    kind,
    id,
    username,
    <comment>);
```

Think it would help the meaning shine through, rather than be somewhat clouded by the mechanics of immutable obj construction.